test/type_construction_test.rb

```diff
@@ -6950,6 +6950,62 @@ def m: [T] () { (untyped) -> T } -> T
     end
   end
 
+  def test_block_pass_and_method__valid
+    with_checker(<<-RBS) do |checker|
+class Object
+  def method: (Symbol) -> Method
+  def my_to_s: (Integer) -> String
+end
+
+class Method
+end
+    RBS
+      source = parse_ruby(<<-EOF)
+x = [1, 2, 3]
+r = x.map(&method(:my_to_s))
+      EOF
+
+      with_standard_construction(checker, source) do |construction, typing|
+        pair = construction.synthesize(source.node)
+
+        assert_no_error typing
+
+        assert_equal parse_type("::Array[::String]"), pair.context.type_env[:r]
+      end
+    end
+  end
+
+  def test_block_pass_and_method__invalid
+    with_checker(<<-RBS) do |checker|
+class Object
+  def method: (Symbol) -> Method
+  def my_to_s: (String) -> String
+end
+
+class Method
+end
+    RBS
+      source = parse_ruby(<<-EOF)
+x = [1, 2, 3]
+r = x.map(&method(:my_to_s))
+      EOF
+
+      with_standard_construction(checker, source) do |construction, typing|
+        pair = construction.synthesize(source.node)
+
+        assert_typing_error typing, size: 1 do |errors|
+          assert_any!(errors) do |error|
+            assert_instance_of Diagnostic::Ruby::BlockTypeMismatch, error
+            assert /^\^\(::Integer\) -> X\(\d+\)$/ =~ error.expected.to_s
```

Comment:
I use regexp to check the BlockTypeMismatch#expected. Is there a better way?

Reply:
Ah, in fact, you may want to use the newer `type_check_test.rb`.